src/status_im/ui/screens/accounts/events.cljs

```diff
@@ -96,11 +96,16 @@
 
 (register-handler-fx
   :add-account
-  (fn [{{:keys [network] :as db} :db} [_ {:keys [address] :as account} password]]
-    (let [account' (assoc account :network network)]
-      {:db             (assoc-in db [:accounts/accounts address] account')
-       ::save-account  account'
-       :dispatch-later [{:ms 400 :dispatch [:login-account address password true]}]})))
+  (fn [{{:keys          [network]
+         :networks/keys [networks]
+         :as            db} :db} [_ {:keys [address] :as account} password]]
+    (let [account' (assoc account :network network
+                                  :networks networks)]
+      (merge
+        {:db            (assoc-in db [:accounts/accounts address] account')
+         ::save-account account'}
+        (when password
+          {:dispatch-later [{:ms 400 :dispatch [:login-account address password true]}]})))))
```

Comment:
What's the reason for this delayed dispatch ?
And why do we actually use dispatch at all, instead of merging effects of `:login-account` handler into `:add-account` effects ?

Reply:
https://github.com/status-im/status-react/pull/1756#pullrequestreview-61658341

Reply:
I looked at the commentary, but I still think what we are doing there (doing some "that should be enough" delay to solve race conditions) is fundamentally wrong. 
When the problem is that effect (adding realm account and switching to it in this case) has to be already performed in order for next action (login process for the new account) to be valid, right solution is to attach callback (or better `:next-event`) to `::save-account` effect and only continue with process once effect takes place.

Reply:
Yep, ok, let's file PR with this change then.

Reply:
Correct, I will be happy to do it.